[WIP] Adds the ability to specify [camelN] as a title-related field marker for citation key generation

MKDJr opened the pull request:
Adds the ability to specify `[camelN]` as a title-related field marker for citation key generation. This is identical to `[camel]` but with a maximum of N words.

Related documentation pull request: https://github.com/JabRef/user-documentation/pull/481 (was https://github.com/JabRef/user-documentation/pull/478 but I changed the fork's branch name).

_This is my first time making a pull request that is not just a typo fix, so any feedback is welcomed._ : )

### Mandatory checks

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

Siedlerchr commented:
Thank you very much for your contribution, I like the idea, but however, I think instead of arbitrarily limiting it to 7 words, it may be useful to specify a parameter with the number of characters, e.g. how it's done by `keywordN`

Siedlerchr commented on src/main/java/org/jabref/logic/citationkeypattern/BracketedPattern.java:
```diff
@@ -662,6 +664,37 @@ private static String camelizeTitle(String title) {
+        int count = 0;
+        try (Scanner titleScanner = new Scanner(formattedTitle)) {
+            while (titleScanner.hasNext()) {
+                if (count >= 7) {break}
```
```suggestion
                if (count >= 7) { break; }
```

Siedlerchr commented:
@MKDJr  It would be really nice if you could continue workin on this feature. Is there anything we can help you with?

MKDJr commented:
I will let you know! Things have been busy this week -- I'd like to keep working on it this weekend.

MKDJr commented:
Finally got around to finishing this request! I set up the dev environment with IntelliJ per the contributing docs. I liked your suggestion of having the pattern work like `keywordN`, `shortCamel` is now `camelN`. I updated the docs too (I submitted a new PR because I changed the forked repo's branch name, so that's linked above).

Siedlerchr commented:
Thanks, that already looks good so far. I will take a closer look later. +1 for adding a test!
